tasks/src/tasks/authenticate.rs

```diff
@@ -67,27 +92,47 @@ impl Task for AuthenticateTask {
 
     fn process_continuation_request_authenticate(
         &mut self,
-        _: CommandContinuationRequest<'static>,
+        continuation: CommandContinuationRequest<'static>,
     ) -> Result<AuthenticateData<'static>, CommandContinuationRequest<'static>> {
-        if self.ir {
-            Ok(AuthenticateData::Cancel)
-        } else if let Some(line) = self.line.take() {
-            Ok(AuthenticateData::r#continue(line))
+        let data = if self.ir {
+            if let AuthMechanism::XOAuth2 = self.mechanism {
+                // The current continuation request indicates an error,
```

Comment:
Could you explain this a bit more? I feel like I don't understand what is happening here.

From my understanding, the code is describing roughly this trace:

```imap
C: A1 AUTHENTICATE XOAUTH <data>
S: + ... // This must not happen because XOAUTH2 <data> was already send (allowed due to SASL-IR)
// error!("cannot authenticate...")
C: <empty> // line 112
```

Is this what you had in mind? Have you seen this in the real world?

If so, I think the correct approach is to send `AuthenticateData::Cancel`, i.e., 

```imap
C: A1 AUTHENTICATE XOAUTH <data>
S: + ... // This must not happen because XOAUTH2 <data> was already send (allowed due to SASL-IR)
C: * // Here, a single "*" send by the client signals "cancel authentication". It's... IMAP...
```

Reply:
What worries me a bit about sending empty data is that there is a chace we end up in an infinite loop.

```
C: A1 AUTHENTICATE XOAUTH <data>
S: + ... // This must not happen because XOAUTH2 <data> was already send (allowed due to SASL-IR)
// error!("cannot authenticate...")
C: <data> // empty data
S: + ... // This must not happen because XOAUTH2 <data> was already send (allowed due to SASL-IR)
// error!("cannot authenticate...")
C: <data> // empty data
S: + ... // This must not happen because XOAUTH2 <data> was already send (allowed due to SASL-IR)
// error!("cannot authenticate...")
C: <data> // empty data
...
```

Sending `AuthenticateData::Cancel` would avoid the loop.